src/main/java/data/workouts/WorkoutList.java

```diff
@@ -245,7 +318,7 @@ public Workout updateWorkout(String userArgument) throws ArrayIndexOutOfBoundsEx
             throw new InvalidWorkoutException(className, InvalidWorkoutException.INVALID_REPS_VALUE_ERROR_MSG);
         }
 
-        Workout updatedWorkout = workoutsList.get(indexToUpdate - 1);
+        Workout updatedWorkout = getWorkoutFromIndexNum(indexToUpdate);
```

Comment:
On second thoughts, I think this should be called "workoutToUpdate" instead of updatedWorkout because here, the workout value is not yet updated.

Reply:
Discussed with @Musfirahe0556596. The initial suggestion was to create a second variable so that there will be two variables: one to represent the object before updating (`workoutToUpdate`) and another to represent the object after updating (`updatedWorkout`). However, we realised this is redundant as both variables point to the same object that will be updated. Thus, the resolution was to rename the variable to `workoutToUpdate`.

The change is reflected in Commit d3b661a.